common/corerequestcontext.go

```diff
@@ -88,10 +88,15 @@ func UpdateResponseStatus(ctx context.Context, status int) error {
 	return nil
 }
 
-func CoreRequestContextMiddleware() func(next http.Handler) http.Handler {
+func CoreRequestContextMiddleware(logger *logrus.Logger) func(next http.Handler) http.Handler {
```

Comment:
you could make it `logger... *logrus.Logger` if you want it as an optional param which would be fine.